MyListLibrary/LinkedList.cs

```diff
@@ -0,0 +1,412 @@
+        private Node<T> Root { get; set; }
+
+        public T this[int index]
+        {
+            get
+            {
+                if (index < 0 || index >= Count)
+                {
+                    throw new IndexOutOfRangeException();
+                }
+
+                return GetNode(index).Value;
+            }
+            set
+            {
+                if (index < 0 || index > Count)
+                {
+                    throw new IndexOutOfRangeException();
+                }
+
+                GetNode(index).Value = value;
+            }
+        }
+
+        public Node<T> GetNode(int index)
+        {
+            Node<T> temp = Root;
+            for (int i = 0; i < index; i++)
+            {
+                if(temp.Next == null)
+                {
+                    throw new IndexOutOfRangeException();
+                }
+
+                temp = temp.Next;
+            }
+
+            return temp;
+        }
+
+        public void AddByIndex(int index, T value)
+        {
+            if (index < 0 || index > Count)
+            {
+                throw new IndexOutOfRangeException();
+            }
+
+            Node<T> newNode = new(value);
+
+            if (index != Count)
+            {
+                newNode.Next = GetNode(index);
+            }
+
+            if (index != 0)
+            {
+                GetNode(index - 1).Next = newNode;
+            }
+            else
+            {
+                Root = newNode;
+            }
+
+            ++Count;
+        }
+
+        public void AddElementsByIndex(int index, IMyList<T> list)
+        {
+            if (index > Count)
+            {
+                throw new IndexOutOfRangeException();
+            }
+
+            LinkedList<T> insertList;
+
+            if (!(list is LinkedList<T>))
+            {
+                insertList = ToLinkedList(list);
+            }
+            else
+            {
+                insertList = (LinkedList<T>)list;
+            }
+
+            if (Root == null)
+            {
+                Root = insertList.Root;
+            }
+
+            Node<T> curr = GetNode(index);
+
+            if (index != 0)
+            {
+                GetNode(index - 1).Next = insertList.GetNode(0);
+            }
+
+            if (index != Count)
+            {
+                insertList.GetNode(insertList.Count - 1).Next = curr;
+            }
+
+            Count+= list.Count;
+        }
+
+        public void AddElementsFirst(IMyList<T> list)
+        {
+            AddElementsByIndex(0, list);
+        }
+
+        public void AddElementsLast(IMyList<T> list)
+        {
+            AddElementsByIndex(Count, list);
+        }
+
+        public void AddFirst(T value)
+        {
+            AddByIndex(0, value);
+        }
+
+        public void AddLast(T value)
+        {
+            AddByIndex(Count, value);
+        }
+
+        public void EditElementByIndex(int index, T value)
```

Comment:
Already have indexer. Don't require this method

Reply:
fix